form-flow/src/test/kotlin/com/ritense/formflow/service/FormFlowDeploymentServiceIntTest.kt

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2015-2022 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.formflow.service
+
+import com.ritense.formflow.BaseIntegrationTest
+import com.ritense.formflow.repository.FormFlowDefinitionRepository
+import com.ritense.formflow.repository.FormFlowStepRepository
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+
+internal class FormFlowDeploymentServiceIntTest : BaseIntegrationTest() {
+
+    @Autowired
+    lateinit var formFlowService: FormFlowService
+
+    @Autowired
+    lateinit var formFlowDeploymentService: FormFlowDeploymentService
+
+    @Autowired
+    lateinit var formFlowDefinitionRepository: FormFlowDefinitionRepository
+
+    @Autowired
+    lateinit var formFlowStepRepository: FormFlowStepRepository
+
+    @BeforeEach
+    fun beforeEach() {
+        formFlowDeploymentService.deployAll()
+    }
+
+    @AfterEach
+    fun afterEach() {
+        formFlowStepRepository.deleteAll()
+        formFlowDefinitionRepository.deleteAll()
+    }
+
+    @Test
+    fun `should auto deploy Form Flow from config directory`() {
+
+        val inkomensLoket = formFlowService.findLatestDefinitionByKey("inkomens_loket")
+
+        assertThat(inkomensLoket!!).isNotNull
+        assertThat(inkomensLoket.id.key).isEqualTo("inkomens_loket")
+        assertThat(inkomensLoket.id.version).isEqualTo(1L)
+        assertThat(inkomensLoket.startStep).isEqualTo("woonplaats")
+        assertThat(inkomensLoket.steps).hasSize(7)
+    }
+
+    @Test
+    fun `should not deploy same Form Flow twice`() {
+        formFlowDeploymentService.deployAll()
+        formFlowDeploymentService.deployAll()
```

Comment:
The second deployAll() isn't needed here because it's already done in the beforeEach() It might be better to make it more consistent by adding the deploy to the previous test, or removing 1 deployAll in this method.

Reply:
changed it